packages/openneuro-server/graphql/resolvers/dataset.js

```diff
@@ -9,11 +9,25 @@ import { permissions } from './permissions.js'
 import { datasetComments } from './comment.js'
 import * as dataladAnalytics from '../../datalad/analytics.js'
 import DatasetModel from '../../models/dataset.js'
+import fetch from 'node-fetch'
 
-export const dataset = (obj, { id }, { user, userInfo }) => {
-  return checkDatasetRead(id, user, userInfo).then(() => {
-    return datalad.getDataset(id)
-  })
+const checkBrainlife = async datasetId => {
+  try {
+    const url = `https://brainlife.io/api/warehouse/project?find={"openneuro.dataset_id":"${datasetId}"}`
+    const res = await fetch(url)
+    const body = await res.json()
+    return Boolean(body.count)
+  } catch (err) {
+    return false
+  }
+}
+
+export const dataset = async (obj, { id }, { user, userInfo }) => {
+  const dataset = await checkDatasetRead(id, user, userInfo).then(() =>
+    datalad.getDataset(id),
+  )
+  dataset.onBrainlife = await checkBrainlife(id)
+  return dataset
```

Comment:
This forces queries which do not request onBrianlife to wait for the field to resolve and could lead to the field not being populated in some cases. Especially since this is an external service, we want it to only block queries which request it.

If you look at the end of this file, there's an object which defines the map of resolvers. You should just add a property onBrainlife which maps to the checkBrainLife function to fix this.